### PR TITLE
Add Support to Deploy Vmware IPI cluster

### PR DESCRIPTION
**Pull request by prsurve**
Signed-off-by: Pratik Surve <prsurve@redhat.com>

### CODE REVIEW

File: ocs_ci/deployment/vmware.py [hunk 1]
```diff
@@ -783,6 +783,112 @@ def destroy_scaleup_nodes(
...
+            with open(install_config_backup, "w") as f:
+                f.write(install_config_str)
+
+        def deploy(self, log_cli_level="DEBUG"):
+            """
+            Deployment specific to OCP cluster on this platform
+            Args:
+                log_cli_level (str): openshift installer's log level
+                    (default: "DEBUG")
+            """
```
Comment (ebenahar) [inline ocs_ci/deployment/vmware.py:846]: empty lines are missing
```suggestion
            Deployment specific to OCP cluster on this platform
            
            Args:
                log_cli_level (str): openshift installer's log level
                    (default: "DEBUG")
                    
            """
```

Comment (prsurve) [reply ocs_ci/deployment/vmware.py]: Done

**Review by ebenahar — DISMISSED** [review]

File: ocs_ci/deployment/vmware.py [hunk 2]
```diff
@@ -783,6 +783,114 @@ def destroy_scaleup_nodes(
...
+                f.write(install_config_str)
+
+        def deploy(self, log_cli_level="DEBUG"):
+            """
+            Deployment specific to OCP cluster on this platform
+
+            Args:
+                log_cli_level (str): openshift installer's log level
+                    (default: "DEBUG")
+
```
Comment (clacroix12) [inline ocs_ci/deployment/vmware.py:847]: This line apparently contains whitespace and is failing the flake8 checks:

```
ocs_ci/deployment/vmware.py:847:1: W293 blank line contains whitespace
```

**Review by clacroix12 — CHANGES_REQUESTED** [review]

File: ocs_ci/deployment/vmware.py (hunk 2 above)
Comment (prsurve) [reply ocs_ci/deployment/vmware.py]: Done

File: ocs_ci/deployment/vmware.py [hunk 3]
```diff
@@ -783,6 +784,121 @@ def destroy_scaleup_nodes(
...
+        def deploy_prereq(self):
+            """
+            Overriding deploy_prereq from parent. Perform all necessary
+            prerequisites for VSPHEREIPI here.
+            """
+            super(VSPHEREIPI.OCPDeployment, self).deploy_prereq()
+
+        def create_config(self):
+            """
+            Creates the OCP deploy config for the Bare Metal
```
Comment (vavuthu) [inline ocs_ci/deployment/vmware.py:810]:
```suggestion
            Creates the OCP deploy config for the vSphere
```

File: ocs_ci/deployment/vmware.py [hunk 4]
```diff
@@ -783,6 +784,121 @@ def destroy_scaleup_nodes(
...
+    A class to handle vSphere IPI specific deployment
+    """
+
+    def __init__(self):
+        super(VSPHEREIPI, self).__init__()
+
+    class OCPDeployment(BaseOCPDeployment):
+        def __init__(self):
+            super(VSPHEREIPI.OCPDeployment, self).__init__()
+            self.helper_node_details = load_auth_config()["baremetal"]
```
Comment (vavuthu) [inline ocs_ci/deployment/vmware.py:798]: didn't the usage of this anywhere. can we remove this if we are not used

File: ocs_ci/deployment/vmware.py [hunk 5]
```diff
@@ -783,6 +784,121 @@ def destroy_scaleup_nodes(
...
+            )
+            install_config_str = yaml.safe_dump(install_config_obj)
+            install_config = os.path.join(self.cluster_path, "install-config.yaml")
+            install_config_backup = os.path.join(
+                self.cluster_path, "install-config.yaml.backup"
+            )
+            with open(install_config, "w") as f:
+                f.write(install_config_str)
+            with open(install_config_backup, "w") as f:
+                f.write(install_config_str)
```
Comment (vavuthu) [inline ocs_ci/deployment/vmware.py:847]: can we copy the file instead writing twice

**Review by vavuthu — DISMISSED** [review]

File: ocs_ci/deployment/vmware.py (hunk 3 above)
Comment (prsurve) [reply ocs_ci/deployment/vmware.py]: Done

File: ocs_ci/deployment/vmware.py (hunk 5 above)
Comment (prsurve) [reply ocs_ci/deployment/vmware.py]: removed backup file

File: conf/deployment/vsphere/ipi_1az_rhcos_vmfs_3m_3w.yaml [hunk 6]
```diff
@@ -0,0 +1,14 @@
+---
```
Comment (vavuthu) [inline conf/deployment/vsphere/ipi_1az_rhcos_vmfs_3m_3w.yaml:1]: could you change filename to vsan instead of vmfs

Comment (prsurve) [reply conf/deployment/vsphere/ipi_1az_rhcos_vmfs_3m_3w.yaml:1]: Done

**Review by vavuthu — APPROVED** [review]

**Review by vasukulkarni — APPROVED** [review]

**Review by clacroix12 — APPROVED** [review]

**Pull request merged**